File: Java/Question5.java
import java.util.Scanner;

public class Question5 {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int n = scanner.nextInt();
		scanner.nextLine();

		int[][][] arr = new int[n][5][5];

		for (int k = 0; k < n; k++) {
			for (int i = 0; i < 5; i++) {
				String[] row = scanner.nextLine().split(" ");
				for (int j = 0; j < 5; j++) {
					if (row[j].equals("?")) {
						arr[k][i][j] = i + j + 2 * k + 1;
					} else {
						arr[k][i][j] = Integer.parseInt(row[j]);
					}
				}
			}
		}

		int maxK = 0, maxI = 0, maxSum = 0;

		for (int k = 0; k < n; k++) {
			for (int i = 0; i < 5; i++) {
				int currSum = 0;
				for (int j = 0; j < 5; j++) {
					currSum += arr[k][i][j];
				}
				if (currSum > maxSum) {
					maxK = k;
					maxI = i;
					maxSum = currSum;
				}
			}
		}

		System.out.println(maxK + " " + maxI);
	}
}

File: C++/Question 5.cpp
#include <iostream>
#include <string>
#include <numeric>

using namespace std;

int main() {
    int n;
    cin >> n;
    int ans[n][5][5];
    string arr[n][5][5];

    for (int x = 0; x < n; x++) {
        for (int i = 0; i < 5; i++) {
            cin >> arr[x][i][0] >> arr[x][i][1] >> arr[x][i][2] >> arr[x][i][3] >> arr[x][i][4];
        }
    }

    for (int k = 0; k < n; k++) {
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < 5; j++) {
                if (arr[k][i][j] == "?") {
                    ans[k][i][j] = i + j + 2 * k + 1;
                } else {
                    ans[k][i][j] = stoi(arr[k][i][j]);
                }
            }
        }
    }

    int max_k = 0;
    int max_i = 0;
    int max_sum = 0;

    for (int k = 0; k < n; k++) {
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < 5; j++) {
                int curr_sum = 0;
                for(auto& num : ans[k][i]) curr_sum += num;
                if (curr_sum > max_sum) {
                    max_k = k;
                    max_i = i;
                    max_sum = curr_sum;
                }
            }
        }
    }

    cout << max_k << " " << max_i << endl;

    return 0;
}